### PR TITLE
return hub exception only from ToIotHubClientContract methods

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
In the AMQP layer, this PR is for both the hub device and service.

### CODE REVIEW

File: iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs
Comment (brycewang-microsoft) [inline iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:92]: these will be addressed in the "else" block below and returned as `IotHubClientException`.

File: iothub/service/src/Amqp/AmqpClientHelper.cs
Comment (brycewang-microsoft) [inline iothub/service/src/Amqp/AmqpClientHelper.cs:129]: similarly here, these will be addressed in the "else" block below and returned as `IotHubServiceException`.

File: iothub/service/src/Amqp/AmqpClientHelper.cs
Comment (brycewang-microsoft) [inline iothub/service/src/Amqp/AmqpClientHelper.cs:141]: same thing here

File: iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:139]: Does this comment need to move or be removed?

File: iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:214]:
```suggestion
                retException = new("IoT hub {0} is suspended".FormatInvariant(message), IotHubClientErrorCode.Suspended);
```

File: iothub/service/src/Amqp/AmqpClientHelper.cs
Comment (drwill-ms) [inline iothub/service/src/Amqp/AmqpClientHelper.cs:39]: Bummer that we have to create a whole new exception when we have one. Perhaps we pass in the inner to ToIotHubClientContract?

**Review by drwill-ms — APPROVED** [review]

**Comment by drwill-ms** [comment]
Please get sign-off from Tim before merge.

File: iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs
Comment (brycewang-microsoft) [reply iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:139]: It seems that this comment is for several transaction-error-related AmqpErrorCode in the following else-ifs, so no, we can just keep it as-is.

File: iothub/service/src/Amqp/AmqpClientHelper.cs
Comment (brycewang-microsoft) [reply iothub/service/src/Amqp/AmqpClientHelper.cs:39]: Sure, I like this idea!

File: iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs
Comment (timtay-microsoft) [inline iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:66]: @drwill-ms how do you feel about using ```new()``` in contexts like these where the variable's type isn't immediately obvious?

Should it be
```csharp
retException = new IotHubClientException(...);
```
instead?

Comment (drwill-ms) [reply iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs:66]: Yeah, we should basically just use it for class fields where the type must be on the left or for inline init of properties. I agree, in this case it isn't as obvious so we should be explicit.

**Review by timtay-microsoft — APPROVED** [review]
Just the one comment. Otherwise looks good

**Pull request merged**
